File: frontend/src/features/tests/BDITest/types/BDITest.ts
import TestBase from "@/schemas/TestBase";

export default interface BDITest extends TestBase {
    questions: BDIQuestion[];
}

export interface BDIQuestion {
    id: string;
    answers: BDIAnswer[];
}

export interface BDIAnswer {
    name: string;
    mark: number | null;
}

File: old-backend/app/domains/tests/bdi/schemas/bdi_test.py
from pydantic import ConfigDict

from app.domains.tests.base.test_base import TestBase
from app.domains.tests.bdi.schemas.bdi_question import BDIQuestion


class BDITest(TestBase):
    type: str = "bdi"
    questions: list[BDIQuestion]

    model_config = ConfigDict(
        json_schema_extra={
            "example": {
                **TestBase.model_config["json_schema_extra"]["example"],
                "questions": [
                    BDIQuestion.model_config["json_schema_extra"]["example"],
                ]
            }
        }
    )
